src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,59 @@
 package core.basesyntax;
 
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+
 public class WorkWithFile {
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String COMMA = ",";
+    private static final String RESULT = "result";
+    private int sumOfSupply = 0;
+    private int sumOfBuy = 0;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        readDataFromFile(fromFileName);
+        String report = createReport(sumOfSupply, sumOfBuy);
+        writeDataToFile(toFileName, report);
+        resetCalculations();
```

Comment:
The `resetCalculations` method is not needed if `sumOfSupply` and `sumOfBuy` are converted to local variables. The state wouldn't persist across multiple calls to `getStatistic`, eliminating the need for resetting.